google-cloud-spanner-hibernate-testing/src/test/java/org/hibernate/userguide/hql/HQLTest.java

```diff
@@ -0,0 +1,2293 @@
+      Session session = entityManager.unwrap( Session.class );
+      //tag::hql-api-stream-projection-example[]
+      try ( Stream<Object[]> persons = session.createQuery(
+        "select p.name, p.nickName " +
+        "from Person p " +
+        "where p.name like :name" )
+      .setParameter( "name", "J%" )
+      .stream() ) {
+
+        persons
+        .map( row -> new PersonNames(
+            (String) row[0],
+            (String) row[1] ) )
+        .forEach( this::process );
+      }
+      //end::hql-api-stream-projection-example[]
+    });
+  }
+
+  @Test
+  public void test_hql_api_scroll_projection_example() {
+    doInJPA( this::entityManagerFactory, entityManager -> {
+      Session session = entityManager.unwrap( Session.class );
+      //tag::hql-api-scroll-example[]
+      try ( ScrollableResults scrollableResults = session.createQuery(
+          "select p " +
+          "from Person p " +
+          "where p.name like :name" )
+          .setParameter( "name", "J%" )
+          .scroll()
+      ) {
+        while(scrollableResults.next()) {
+          Person person = (Person) scrollableResults.get()[0];
+          process(person);
+        }
+      }
+      //end::hql-api-scroll-example[]
+    });
+  }
+
+  @Test
+  public void test_hql_api_scroll_open_example() {
+    ScrollableResults scrollableResults = doInJPA( this::entityManagerFactory, entityManager -> {
+      Session session = entityManager.unwrap( Session.class );
+      return session.createQuery(
+        "select p " +
+        "from Person p " +
+        "where p.name like :name" )
+      .setParameter( "name", "J%" )
+      .scroll();
+    });
+    try {
+      scrollableResults.next();
+      fail("Should throw exception because the ResultSet must be closed by now!");
+    }
+    catch ( Exception expected ) {
+    }
+  }
+
+  private void process(Person person) {
+  }
+
+  private void process(PersonNames personName) {
+  }
+
+  @Test
+  public void test_hql_api_unique_result_example() {
+    doInJPA( this::entityManagerFactory, entityManager -> {
+      Session session = entityManager.unwrap( Session.class );
+      //tag::hql-api-unique-result-example[]
+      Person person = (Person) session.createQuery(
+        "select p " +
+        "from Person p " +
+        "where p.name like :name" )
+      .setParameter( "name", "J%" )
+      .uniqueResult();
+      //end::hql-api-unique-result-example[]
+    });
+  }
+
+  @Test
+  public void test_hql_string_literals_example_1() {
+    doInJPA( this::entityManagerFactory, entityManager -> {
+      //tag::hql-string-literals-example[]
+      List<Person> persons = entityManager.createQuery(
+        "select p " +
+        "from Person p " +
+        "where p.name like 'Joe'", Person.class)
+      .getResultList();
+      //end::hql-string-literals-example[]
+    });
+  }
+
+  @Test
+  @Ignore
+  // TODO: Test fails due to quote literals in string: 'Joe''s'; figure out if there
```

Comment:
It's not clear in this file which parts are original, and which ones you changed. Maybe you can split your changes into a separate commit?

Also, this TODO should probably be an issue we're tracking.

Finally, did you have to delete any tests like the ones involving `= ANY (SELECT ...)`?

Reply:
I'll make a separate PR with the test annotated with `@Ignore` to visualize changes

Reply:
1. Done.
2. Done. Made this issue to track: https://github.com/GoogleCloudPlatform/google-cloud-spanner-hibernate/issues/55
3. Yes, tests using ANY, ALL, and SOME were deleted.